code/game/objects/structures/grey_autocloner.dm

```diff
@@ -0,0 +1,151 @@
+
+	if(!R.dna)
+		R.dna = new /datum/dna()
+
+	var/mob/living/carbon/human/H = new /mob/living/carbon/human(src)
+	H.set_species(/datum/species/grey) //This is a grey cloner after all. Funnier this way tbh
+	occupant = H
+
+	if(!R.dna.real_name)	//to prevent null names
+		R.dna.real_name = H.real_name
+	else
+		H.real_name = R.dna.real_name
+
+	H.dna = R.dna.Clone()
+
+	for(var/datum/language/L in R.languages)
+		H.add_language(L.name)
+
+	domutcheck(H, MUTCHK_FORCED) //Ensures species that get powers by the species proc handle_dna keep them
+
+	H.dna.UpdateSE()
+	H.dna.UpdateUI()
+
+	H.sync_organ_dna(1) // It's literally a fresh body as you can get, so all organs properly belong to it
+	H.UpdateAppearance()
+
+	H.Paralyse(8 SECONDS)
+
+	clonemind.transfer_to(H) //INSTANTLY INTO THE CLONE
+	H.ckey = R.ckey
+	update_clone_antag(H) //Since the body's got the mind, update their antag stuff right now. Otherwise, wait until they get kicked out (as per the CLONER_MATURE_CLONE business) to do it.
+	var/message
+	message += "<b>Consciousness slowly creeps over you as your body regenerates.</b><br>"
+	message += "<i>So this is what cloning feels like?</i>"
+	to_chat(H, "<span class='notice'>[message]</span>")
+
+	update_icon()
+
+	H.suiciding = FALSE
+	attempting = FALSE
+	addtimer(CALLBACK(src, PROC_REF(finish_clone)), 1 MINUTES)
+	return 1
+
+/obj/machinery/grey_autocloner/process()
+	if(stat & NOPOWER) //explode if power is lost and cloning
+		if(occupant)
+			messy_explode()
+			return
+
+	else if((occupant) && (occupant.loc == src))
+		occupant.Paralyse(8 SECONDS)
+		use_power(15000) //Hope the room has power!
+		for(var/bt in brine_types)
+			if(occupant.reagents.get_reagent_amount(bt) < 1)
+				occupant.reagents.add_reagent(bt, 1)
+
+/obj/machinery/grey_autocloner/proc/update_clone_antag(mob/living/carbon/human/H)
+	// Check to see if the clone's mind is an antagonist of any kind and handle them accordingly to make sure they get their spells, HUD/whatever else back.
+	if(H.mind in SSticker.mode.syndicates)
+		SSticker.mode.update_synd_icons_added()
+	if(H.mind in SSticker.mode.cult)
+		SSticker.mode.update_cult_icons_added(H.mind) // Adds the cult antag hud
+		SSticker.mode.add_cult_actions(H.mind) // And all the actions
+		if(SSticker.mode.cult_risen)
+			SSticker.mode.rise(H)
+			if(SSticker.mode.cult_ascendant)
+				SSticker.mode.ascend(H)
+
+/obj/machinery/grey_autocloner/proc/finish_clone()
+	playsound(src.loc, 'sound/machines/ding.ogg', 50, 1)
+	var/obj/item/organ/internal/storedorgan = new /obj/item/organ/internal/cyberimp/brain/speech_translator
+	storedorgan.insert(occupant) //insert stored organ into the user
+	message_admins("fries are done")
+	occupant.forceMove(get_turf(src))
+	occupant.update_body()
+	domutcheck(occupant) //Waiting until they're out before possible notransform.
+	occupant.special_post_clone_handling()
+	occupant = null
+	update_icon()
+
+/obj/machinery/grey_autocloner/emp_act(severity)
+	messy_explode()
```

Comment:
uhhh you sure you want this to just explode if it gets graced by any emp?

Reply:
Yes